File: app/src/main/java/com/nfcalarmclock/NacPreferenceCategory.java
package com.nfcalarmclock;

import android.content.Context;
//import android.preference.PreferenceCategory;
import android.util.AttributeSet;
import android.view.View;
import android.widget.TextView;

import androidx.preference.PreferenceCategory;
import androidx.preference.PreferenceViewHolder;

/**
 * Preference category.
 */
public class NacPreferenceCategory
	extends PreferenceCategory
{

	/**
	 */
	public NacPreferenceCategory(Context context)
	{
		super(context);
	}

	/**
	 */
	public NacPreferenceCategory(Context context, AttributeSet attrs)
	{
		super(context, attrs);
	}

	/**
	 */
	public NacPreferenceCategory(Context context, AttributeSet attrs, int style)
	{
		super(context, attrs, style);
	}

	/**
	 */
	@Override
	public void onBindViewHolder(PreferenceViewHolder holder)
	{
		super.onBindViewHolder(holder);

		Context context = getContext();
		NacSharedPreferences shared = new NacSharedPreferences(context);
		TextView title = (TextView) holder.findViewById(android.R.id.title);

		title.setTextColor(shared.getThemeColor());
	}

}

File: app/src/main/java/com/nfcalarmclock/settings/preference/NacPreferenceCategory.kt
package com.nfcalarmclock.settings.preference

import android.R
import android.content.Context
import android.util.AttributeSet
import android.widget.TextView
import androidx.preference.PreferenceCategory
import androidx.preference.PreferenceViewHolder
import com.nfcalarmclock.shared.NacSharedPreferences

/**
 * Preference category.
 */
class NacPreferenceCategory
	: PreferenceCategory
{

	/**
	 * Shared preferences.
	 */
	private var sharedPreferences: NacSharedPreferences = NacSharedPreferences(context)

	/**
	 * Constructor.
	 */
	constructor(context: Context) : super(context)
	{
	}

	/**
	 * Constructor.
	 */
	constructor(context: Context, attrs: AttributeSet?) : super(context, attrs)
	{
	}

	/**
	 * Constructor.
	 */
	constructor(context: Context, attrs: AttributeSet?, style: Int) : super(context, attrs, style)
	{
	}

	/**
	 * Constructor.
	 */
	init
	{
		isIconSpaceReserved = false
	}

	/**
	 * Called when the view holder is bound.
	 */
	override fun onBindViewHolder(holder: PreferenceViewHolder)
	{
		// Super
		super.onBindViewHolder(holder)

		// Get the title text view
		val title = holder.findViewById(R.id.title) as TextView

		// Set the title text color
		title.setTextColor(sharedPreferences.themeColor)
	}
}